Homework_10_Java_Software_Messenger_Development/Messenger/src/Main.java

```diff
@@ -0,0 +1,118 @@
+		konrad.addContact(hanna);
+
+		System.out.println("Hanna's contacts: ");
+		for (User contact : hanna.getContacts()) {
+			System.out.println(contact.getName());
+		}
+
+		System.out.println("\nBartek's contacts: ");
+		for (User contact : bartek.getContacts()) {
+			System.out.println(contact.getName());
+		}
+
+		System.out.println("\nKonrad's contacts: ");
+		for (User contact : konrad.getContacts()) {
+			System.out.println(contact.getName());
+		}
+
+		System.out.println("\nSending messages: ");
+
+		Message message1 = new Message(hanna, bartek, "Hi, Bartek! ");
+		hanna.sendMessage(message1);
+
+		Message message2 = new Message(bartek,hanna, "Hi, Hanna :) ");
+		bartek.sendMessage(message2);
+
+		Message message3 = new Message(hanna,konrad, "Hi, Konrad! ");
+		hanna.sendMessage(message3);
+
+		System.out.println("\nHanna's messages: ");
+		for (Message message : hanna.getMessages()) {
+			System.out.println(message.getText() + "(Status: " + message.getStatus() + ")");
+		}
+
+		System.out.println("\nBartek's messages: ");
+		for (Message message : bartek.getMessages()) {
+			System.out.println(message.getText() + " (Status: " + message.getStatus() + ")");
+		}
+
+		System.out.println("\nKonrad's messages: ");
+		for (Message message : konrad.getMessages()) {
+			System.out.println(message.getText() + " (Status: " + message.getStatus() + ")");
+		}
+
+		System.out.println("\nReading messages...");
+		bartek.readMessage(message1);
+		hanna.readMessage(message2);
+
+
+		System.out.println("\nHanna's messages after reading: ");
+		for (Message message : hanna.getMessages()) {
+			System.out.println(message.getText() + "(Status: " + message.getStatus() + ")");
+		}
+
+		System.out.println("\nBartek's messages after reading: ");
+		for (Message message : bartek.getMessages()) {
+			System.out.println(message.getText() + " (Status: " + message.getStatus() + ")");
+		}
+
+		System.out.println("\nKonrad's messages not reading: ");
+		for (Message message : konrad.getMessages()) {
+			System.out.println(message.getText() + " (Status: " + message.getStatus() + ")");
+		}
+
+		hanna.removeContact(konrad);
+		System.out.println();
+		System.out.println(hanna.getName() + " removing contact: " + konrad.getName());
+		System.out.println("\nHanna's contacts after removing Konrad: ");
+		for (User contact : hanna.getContacts()) {
+			System.out.println(contact.getName());
+		}
+
+		hanna.deleteMessage(message1);
+		System.out.println();
+		System.out.println(hanna.getName() + " deleting a message from: " + bartek.getName());
+		System.out.println("\nHanna's messages after deleting a message: ");
+		for (Message message : hanna.getMessages()) {
+			System.out.println(message.getText() + " (Status: " + message.getStatus() + ")");
+		}
+
+		List<User> users = Arrays.asList(hanna, bartek, konrad);
+
+		/**
+		 * Search for messages by RECEIVED status
+		 */
+		List<Message> receivedMessages = User.findMessagesByStatus(users, MessageStatus.RECEIVED);
```

Comment:
It's a way, but we're imitating some business logic here, right :)

There might be a security incident as ideally we don't want to allow other users or anybody except user itself to read/list/filter their own messages.

I mean this functionality can be a part of the `SystemAdministrator` access but we don't have such entity in out program and if Bartek doesn't want Hanna to read his messages - such static method either shouldn't exist or should be accessible only for Admin of the system (which was not required here ofc)